addon/-debug/utils/validation-decorator.js

```diff
@@ -55,7 +55,7 @@ EmberObject.reopenClass({
         typeRequired
       } = validations[key];
 
-      if (isRequired && !instance.hasOwnProperty(key)) {
+      if (isRequired && instance[key] === undefined && !instance.hasOwnProperty(key)) {
```

Comment:
Hmm, why do we check for both here? I would have expected that you would remove the `!instance.hasOwnProperty(key)` part...

Reply:
This is an edge case that I'm not entirely sure about, so I'd be open to removing it if you think it makes sense, but it's to cover the case where someone knowingly passes `undefined` to a component and intends it to be the value of the field.

On the one hand, I do think this is unlikely, and may actually be better to safeguard against. On the other, it seems strange to arbitrarily say you can't pass in a particular value. It would simplify the code if we went that route though, quite a bit too (this pops up in a lot of places, see `defaultIfUndefined`)

Reply:
Worth noting, I think `undefined` meaning "no value was passed in" is also more semantically correct, though I've seen people do crazy weird things with values and types in JS so I was erring on the conservative side here 😛

Reply:
I guess I was just going on the PR description: my understanding from that was that if a parent class provides a field value (even if it is `undefined`) it should be allowed.  This implementation does not allow that...

If we want to move to allow anywhere on the prototype to have a value, then we would change the guard to `isRequired && key in instance`...

Reply:
So I do think that would be valid, but currently the following will always define a property on the prototype:

```js
class Foo {
  @required prop;
}
```

This only happens because the decorator was applied to the property and when being applied, defines the property on the prototype if no initializer exists. I'm unsure of what the spec says should happen here, but I think there's a strong chance this is just something the Babel transform is doing that will change in upcoming versions.

I could go either way here now with that realization. We could be less restrictive and allow `undefined` on the instance, even if we can't allow it on the prototype chain, or we could be more consistent and not allow it anywhere until/unless the transforms change. What do you think?